ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Genesis/Governor.hs

```diff
@@ -127,6 +128,30 @@ reprocessLoEBlocksOnCandidateChange chainDb getCandidates =
         reprocessLoEAsync chainDb
         spin newTips
 
+runLoEUpdaterOnChange ::
+  (Monoid a, Eq a, IOLike m, LedgerSupportsProtocol blk) =>
+  UpdateLoEFrag m blk ->
+  m ( UpdateLoEFrag m blk
+    , ChainDB m blk ->
+      STM m a -> -- Re-run the LoE updater whenever this changes
+      m Void
+    )
+runLoEUpdaterOnChange loeUpdater = do
+    varLoEFrag <- newTVarIO $ AF.Empty AF.AnchorGenesis
+    let newUpdateLoEFrag = UpdateLoEFrag $ \_ _ -> readTVarIO varLoEFrag
+    pure (newUpdateLoEFrag, spin varLoEFrag mempty)
+  where
+    spin varLoEFrag oldTrigger chainDb getTrigger = do
+      (newTrigger, curChain, curLedger) <- atomically $ do
+        (_, newTrigger) <- blockUntilChanged id oldTrigger getTrigger
+        curChain <- ChainDB.getCurrentChain chainDb
+        curLedger <- ChainDB.getCurrentLedger chainDb
+        pure (newTrigger, curChain, curLedger)
+      loeFrag <- updateLoEFrag loeUpdater curChain curLedger
+      atomically $ writeTVar varLoEFrag loeFrag
+      reprocessLoEAsync chainDb
```

Comment:
`reprocessLoEAsync` is a bit confusing now. The LoE fragment has been already updated. Maybe this is meant to update the chain selection instead (?)

Reply:
Yeah, `reprocessLoEAsync` is maybe a somewhat misleading name, maybe `triggerChainSelection` is more intuitive?

Reply:
It's reprocessing blocks postponed by the LoE, which I perceive as the overarching mechanism of which the LoE fragment is one part...so it should probably be `reprocessLoEBlocksAsync`?

`triggerChainSelection` seems a bit too general, but maybe it actually is, since we're processing all blocks that can possibly be eligible for extending the chain?

Reply:
I like `triggerChainSelection`.